database.go

```diff
@@ -87,6 +102,39 @@ func (d *Database) GetDocument(ctx context.Context, id string) (Document, error)
 	return doc, err
 }
 
+func (d *Database) GetDocumentVersions(ctx context.Context, id string, withContent bool) ([]Document, error) {
+	var docs []Document
+	var sqlString string
+	if withContent {
+		sqlString = "SELECT id, version, content, language FROM documents where id = $1 ORDER BY version DESC"
+	} else {
+		sqlString = "SELECT id, version FROM documents where id = $1 ORDER BY version DESC"
+	}
+	err := d.SelectContext(ctx, &docs, sqlString, id)
+	return docs, err
+}
+
+func (d *Database) GetDocumentByVersion(ctx context.Context, id string, version int64) (Document, error) {
+	var doc Document
+	err := d.GetContext(ctx, &doc, "SELECT * FROM documents WHERE id = $1 AND version = $2", id, version)
+	return doc, err
+}
+
+func (d *Database) DeleteDocumentByVersion(ctx context.Context, version int64, documentID string) error {
+	res, err := d.ExecContext(ctx, "DELETE FROM documents WHERE version = $1 AND id = $2", version, documentID)
```

Comment:
should be `DeleteDocumentByVersion(ctx context.Context, documentID string, version int64)`